spec/user_spec.rb

```diff
@@ -7,4 +7,9 @@
   end
 
   # Write a new unit test here.
-end
\ No newline at end of file
+  it 'can give new user name' do
+    user = User.new(name: "Amber")
+    expect(user[:name]).to eq "Amber"
```

Comment:
On objects you can access the attributes as methods. `user.name` is cleaner than `user[:name]` Also, you already have a User object stored in `subject` perhaps it would be better to update that object with the name 'Amber'?

Reply:
When I try to change it to ```subject.update(name: "Amber"```, it gives me this error: ```User#update cannot be called on a new resource```

Reply:
Use a setter method and then subject.save

Reply:
Do you mean something like this? 
```
  it 'can give new user name' do
    subject.update(name: "Amber")
    subject.save
    expect(subject.name).to eq "Amber"
  end
```
It still gives me the same error. I feel a bit lost.